geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java

```diff
@@ -353,7 +356,8 @@ public void begin() {
             "Current thread has paused its transaction so it can not start a new transaction");
       }
     }
-    TXId id = new TXId(this.distributionMgrId, this.uniqId.incrementAndGet());
+    TXId id = new TXId(this.distributionMgrId,
+        this.uniqId.updateAndGet(i -> i == Integer.MAX_VALUE ? 0 : i + 1));
```

Comment:
Please note 0 was not a valid TXId before -- due to the uniqId.incrementAndGet() method call. It would be better to wrap to 1 instead of 0.

Reply:
Thanks for your comments. You are right, I have updated the code to wrap to 1.